app.go

```diff
@@ -449,22 +456,46 @@ func ValidateApp(opts ...Option) error {
 	return app.Err()
 }
 
+// logBuffer will buffer all messages until a logger has been
+// initialized.
+type logBuffer struct {
+	entries []fxlog.Entry
+	logger  fxlog.Logger
```

Comment:
As I read it, the value of logBuffer is to store log entries to write before the constructed structured logger is initialized so that those entries can be logged with the structure logger as opposed to another default logger for a more consistent experience.

Is this logger field within logBuffer required? Having a buffer with a logger allows someone to log with both the 'app.log' when its setup as well as the logBuffer once' it's connected, where ideally there should only be a single choice?

The alternative approach would be to replace Connect with a `func (l *logBuffer) Sync(logger fxlog.Logger)`
that only flushes and does not set a logger. Once the structured logger is available, the buffer shouldn't be used anymore? Is there a strong argument against going the other way and even closing off the logBuffer from further writes once its Synced?